packages/wallet/src/services/DelegationTracker/DelegationTracker.ts

```diff
@@ -68,6 +90,95 @@ export const certificateTransactionsWithEpochs = (
     )
   );
 
+export const createDelegationPortfolioTracker = ({
+  rewardAccounts$,
+  knownAddresses$,
+  utxoTracker
+}: DelegationPortfolioTrackerProps): Observable<Cip17Pool[]> => {
+  const balanceAllUtxosTracker = createBalanceByAddressTracker(utxoTracker);
+
+  const delegatedAccounts$ = rewardAccounts$.pipe(
+    map((rewardsAccounts) =>
+      rewardsAccounts.filter(
+        (account) => account.keyStatus === Cardano.StakeKeyStatus.Registered && account.delegatee?.nextNextEpoch
+      )
+    )
+  );
+
+  const hydratedDelegatedAccounts$ = delegatedAccounts$.pipe(
+    withLatestFrom(knownAddresses$),
```

Comment:
I think this is technically a race condition
```suggestion
    blockingWithLatestFrom(knownAddresses$),
```

Reply:
I was counting on knownAddresses to be a TrackingSubject, which would make it repeat any previous value it emitted

Reply:
Right, rewardAccounts$ won't emit until there are knownAddresses$. That's an implementation detail, that's what I meant by 'technically'. Fine to leave as-is.

Reply:
Todo: do not export `createDelegationPortfolioTracker`

Reply:
[78fb90f](https://github.com/input-output-hk/cardano-js-sdk/pull/735/commits/78fb90f4eb7a044097ea5f68fb734731283a8b6d) & [63e7597](https://github.com/input-output-hk/cardano-js-sdk/pull/735/commits/63e7597dea79efdd27dcf138d894f2b6801d4276): symbol is now exported only for unit testing.